packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -104,13 +183,15 @@ class CupertinoTabScaffold extends StatefulWidget {
   /// that lets the user switch between different tabs in the main content area
   /// when present.
   ///
-  /// Setting and changing [CupertinoTabBar.currentIndex] programmatically will
-  /// change the currently selected tab item in the [tabBar] as well as change
-  /// the currently focused tab from the [tabBuilder].
-
+  /// The [CupertinoTabBar.currentIndex] property of [tabBar] is *overridden*
+  /// by the [CupertinoTabController.currentIndex] value in [controller] if [controller]
+  /// is not null or there's an existing controller in this [CupertinoTabScaffold]'s
+  /// internal state object. Otherwise, the scaffold will use [CupertinoTabBar.currentIndex]
```

Comment:
If no [controller] is provided, the scaffold defaults to the [tabBar]'s [CupertinoTabBar.currentIndex] as its initial index. But subsequent changes to [CupertinoTabBar.currentIndex] will not affect the tab scaffold.